src/core/render.ts

```diff
@@ -1,57 +1,232 @@
 /// <reference path="../../defs/es6-promise.d.ts" />
 
 import {Scene} from './scene';
+import {Item} from './items/item';
+import {Source} from './source/source';
 import {App as iApp} from '../internal/app';
 import {Render as iRender} from '../internal/render';
 import { exec } from '../internal/internal';
 
 export class Render {
+  private static _canvasCounter: number = 0;
+  private _index: number;
+  private _fps: number;
+  private _canvas: HTMLCanvasElement;
+  private _rendering: boolean;
+  private static _canvasPool: Render[] = [];
 
+  constructor(props?) {
+    this._index = props.canvasIndex;
+    this._fps = props.fps;
+    this._canvas = props.canvas;
+    this._rendering = props.rendering;
+  }
 
-  // Assign a canvas and create necessary webgl context
-  static setTargetCanvas(canvas: HTMLCanvasElement,
-    fps: number = 30): Promise<any> {
+  static getPreparedCanvases() {
+    return new Promise(resolve => {
+      resolve(Render._canvasPool)
+    })
+  }
+
+  /**
+   * param: (canvas: HTMLCanvasElement, fps?: number)
+   *
+   * Prepare a canvas for rendering.
+   *
+   */
+  static prepareCanvas(canvas: HTMLCanvasElement, fps: number = 30, includeToPool: boolean = true) {
+    return new Promise((resolve, reject) => {
+      const canvasProps = {
+        'canvas': canvas,
+        'fps': fps,
+        'canvasIndex': Render._canvasCounter,
+        'rendering': false
+      };
+      if (includeToPool) {
+        Render._canvasPool.push(new Render(canvasProps))
+      }
+      iRender.initializeCanvas(canvas, Render._canvasCounter, fps).then(() => {
+        resolve(true);
+      })
+      Render._canvasCounter = Render._canvasCounter + 1;
```

Comment:
Not sure if this is intentional. Do we want to increment the canvas counter only if `initializeCanvas` successfully resolved?